[No merge] Release v1.14 rocm

ipanfilo opened the pull request:
# Description

Please include a brief summary of the changes, relevant motivation and context.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

ptrendx committed:
Code drop for version 1.14

Signed-off-by: Przemek Tredak <ptredak@nvidia.com>

ipanfilo committed:
Merge remote-tracking branch 'upstream/release_v1.14' into release_v1.14_rocm

ipanfilo committed:
Resolve build failures

ipanfilo committed:
Revert c2c3d540b1eca9ccbcc0fa7cb871688814a536f9 and b898cbe18fb1d3414544bf8d11de5f83cadfb5db to match release branch baseline
Fix pytorch fused_attn_with_cp tests
Fix CTests

ipanfilo committed:
Fix JAX tests: use bottom-right mask as default

wenchenvincent commented on transformer_engine/jax/attention.py:
```diff
@@ -68,7 +71,9 @@ def is_bottom_right(self):
             AttnMaskType.PADDING_CAUSAL_BOTTOM_RIGHT_MASK,
-        ]
+        ] or ( #Use bottom-right if no mask is set to match TE FusedAttn (CK) code
+            is_hip_extension() and self == AttnMaskType.NO_MASK
```
I thought top-left or bottom-right is only valid for the scenarios where there is a mask (causal or sliding window). Does top-left or bottom-right mean when there is no mask?

ipanfilo replied:
This method is called from https://github.com/ROCm/TransformerEngine/pull/178/commits/2f85f5f23f3fcdf908b595854a6dd80fa4af1236#diff-d6e9c5eeb7a2c944a7123af74042744a9b8dd78234b40942ce76ae71d45ad02aR154 to determine which mask to generate for SWA. So the change means that default mask (when no type is specified) will be bottom-right.

VeeraRajasekhar committed:
Fixed hipblaslt precision issue with mi300 on rocm6.4 & hipblaslt12 docker (#202)

VeeraRajasekhar committed:
[CI] deprecate praxis installation and tests

- Removed praxis installation and related test setup from `ci/jax.sh`
- Installed `flax>=0.7.1`, with typing_extensions>=4.12.2
